src/main/java/org/opensearch/ad/transport/AnomalyResultTransportAction.java

```diff
@@ -558,50 +552,70 @@ private void handleException(Exception e) {
 
             if (featureOptional.getUnprocessedFeatures().isPresent()) {
                 featureInResponse = ParseUtils.getFeatureData(featureOptional.getUnprocessedFeatures().get(), detector);
-                AtomicReference<Exception> failure = new AtomicReference<>();
-                AtomicInteger responseCount = new AtomicInteger();
-                transportService
-                    .sendRequest(
-                        thresholdNode,
-                        EntityResultAction.NAME,
-                        new EntityResultRequest(
-                            detector.getDetectorId(),
-                            Collections
-                                .singletonMap(
-                                    Entity.createEntityByReordering(Collections.emptyMap()),
-                                    featureOptional.getUnprocessedFeatures().get()
-                                ),
-                            dataStartTime,
-                            dataEndTime
-                        ),
-                        option,
-                        new ActionListenerResponseHandler<>(
-                            new EntityResultListener(
-                                thresholdNode.getId(),
-                                detector.getDetectorId(),
-                                failure,
-                                1,
+            }
+
+            if (!featureOptional.getProcessedFeatures().isPresent()) {
+                Optional<Exception> exception = coldStartIfNoCheckPoint(detector);
+                if (exception.isPresent()) {
+                    listener.onFailure(exception.get());
+                    return;
+                }
+
+                if (!featureOptional.getUnprocessedFeatures().isPresent()) {
+                    // Feature not available is common when we have data holes. Respond empty response
+                    // so that alerting will not print stack trace to avoid bloating our logs.
```

Comment:
Not quite get what does this line mean. You mean alerting plugin may print too much log if we don't return empty response?

Reply:
This is from the first PRs and I never changed.  Changed to:

// Feature not available is common when we have data holes. Respond empty response
// and don't log to avoid bloating our logs.